db/migrate/20200922130419_add_user_schools.rb

```diff
@@ -0,0 +1,37 @@
+class AddUserSchools < ActiveRecord::Migration[6.0]
+  def up
+    create_table :user_schools do |t|
+      t.references :user
+      t.references :school
+      t.timestamps
+    end
+
+    populate_table_sql = <<~SQL
+      INSERT INTO user_schools(user_id, school_id, created_at, updated_at)
+      SELECT      id, school_id, NOW(), NOW()
+      FROM        users
+      WHERE       school_id IS NOT NULL
+    SQL
+    connection.execute populate_table_sql
+
+    add_index :user_schools, %i[user_id school_id], unique: true
+    add_index :user_schools, %i[school_id user_id], unique: true
+
+    remove_column :users, :school_id
```

Comment:
I'd suggest we do this in a subsequent change....

Reply:
the only issue then is that ActiveRecord will read it as an attribute, and we may get conflicts with the overridden `school_id` method.

how about renaming it, to `legacy_school_id` or something similar?

Reply:
👍  to rename...

Reply:
done in cc596cf